docs/protocol-extension.md

```diff
@@ -1,6 +1,8 @@
 # GraphQL over WebSocket Protocol extension
 
-The GraphQL over WebSocket Protocol used by `mercurius` follows apollo's protocol defined [here](https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md). In order to achieve gateway-to-service communication and handle `connection_init` per client, an extension is used on the protocol for server-to-server communication.
+The GraphQL over WebSocket Protocol `mercurius` uses by default is [graphql-transport-ws](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md). It also supports the protocol used by Apollo's `subscriptions-transport-ws`; however that library is **unmantained** so it's not recommended to use.
+
+In order to achieve gateway-to-service communication and handle `connection_init` per client, an extension is used on the protocol for server-to-server communication. See https://github.com/mercurius-js/mercurius/issues/268 for more details about the original issue and the actual implemented solution.
```

Comment:
Which clients support WS as a complete transport? We should list compatible clients here.

Reply:
Well, as far as I know: apollo, urql, react-query, graphql-ws... For example, graphql-hooks doesn't support this use case, but I was thinking to make a PR to add support for it after this PR would land 😄 

Anyways, this document refers to a specific use case for the `extension` field, and was added on #271: my changes are not related to this PR in reality (I just fixed/improved it a little bit because it was outdated/unclear).

Reply:
Ok I'll convert this doc into "GraphQL over WebSocket" so we can add more related sections (extensions, supported clients, etc).